addons/account/models/account_move.py

```diff
@@ -2654,8 +2654,8 @@ def copy_data(self, default=None):
             user_fiscal_lock_date = move.company_id._get_user_fiscal_lock_date()
             if (default_date or move.date) <= user_fiscal_lock_date:
                 vals['date'] = user_fiscal_lock_date + timedelta(days=1)
-        if not self.journal_id.active and 'journal_id' in vals_list:
-            del vals['journal_id']
+            if not move.journal_id.active and 'journal_id' in vals_list:
```

Comment:
Use `vals` instead of `valse_list` and add a test case for this.

Reply:
Hello :wave: 
Requested changes are done.
Thanks!